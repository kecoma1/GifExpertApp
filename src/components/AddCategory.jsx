import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('Look for a gif brah');

    // Handling the input
    const handleInputChange = e => setInputValue(e.target.value);

    // Preventing the default behavior of the submit
    const handleSubmit = e => {
        e.preventDefault();
    
        if (inputValue.trim().length > 2) {
            setCategories( c => [inputValue, ...c] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>AddCategory</h2>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
