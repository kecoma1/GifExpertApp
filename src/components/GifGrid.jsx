import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading</p> }

            <div className='card-grid'>
                {
                    data.map( img => 
                        <GifGridItem 
                        key={img.id}
                        { ...img }/>
                    )
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
