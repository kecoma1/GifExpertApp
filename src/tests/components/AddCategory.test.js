import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Tests on <AddCategory />:', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    });

    test('Should display the component correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should change the textbox', () => { 
        const input = wrapper.find('input');
        const value = 'Hola Mundo'

        input.simulate('change', {
            target: {
                value
            }
        });
    });

    test('Should NOT post information on the submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled()
    });

    test('Should call the setCategories and print the textbot', () => {
        const value = 'Hola test'

        // Introducing the input
        wrapper.find('input').simulate('change', { target: { value }});

        // Submitting
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).toHaveBeenCalled();
        expect( wrapper.find('input').prop('value') ).toBe('');
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) );
    });

});