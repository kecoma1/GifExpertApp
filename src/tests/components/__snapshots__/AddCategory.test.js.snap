// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AddCategory />: Should display the component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    AddCategory
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
